libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/app/typing_indicator.py

```diff
@@ -18,61 +18,44 @@
 class TypingIndicator:
     """
     Encapsulates the logic for sending "typing" activity to the user.
+
+    Scoped to a single turn of conversation with the user.
     """
 
-    def __init__(self, intervalSeconds=1) -> None:
-        self._intervalSeconds = intervalSeconds
-        self._task: Optional[asyncio.Task] = None
-        self._running: bool = False
-        self._lock = asyncio.Lock()
-
-    async def start(self, context: TurnContext) -> None:
-        async with self._lock:
-            if self._running:
-                return
-
-            logger.debug(
-                f"Starting typing indicator with interval: {self._intervalSeconds} seconds"
-            )
-            self._running = True
-            self._task = asyncio.create_task(self._typing_loop(context))
-
-    async def stop(self) -> None:
-        async with self._lock:
-            if not self._running:
-                return
-
-            logger.debug("Stopping typing indicator")
-            self._running = False
-            task = self._task
-            self._task = None
-
-        # Cancel outside the lock to avoid blocking
-        if task and not task.done():
-            task.cancel()
-            try:
-                await task
-            except asyncio.CancelledError:
-                pass
-
-    async def _typing_loop(self, context: TurnContext):
-        """Continuously send typing indicators at the specified interval."""
+    def __init__(self, context: TurnContext, interval: float = 10.0) -> None:
+        self._context: TurnContext = context
+        self._interval: float = interval
```

Comment:
Because I'm paranoid, I would also put a bounds check / limit check on the interval. For example, a negative number, or something longer than a day, or whatever limits you think would be absurd. In C#, I would make this an unsigned short, but in python I would just add the bound checking.